cpp/include/raft/lap/lap_kernels.cuh

```diff
@@ -48,20 +48,18 @@ const int AUGMENT{4};
 const int MODIFIED{5};
 
 template <typename weight_t>
-bool __device__ near_zero(weight_t w) {
-  return (w == 0);
+bool __device__ near_zero(weight_t w, weight_t epsilon) {
+  return ((w > -epsilon) && (w < epsilon));
 }
 
 template <>
-bool __device__ near_zero<float>(float w) {
-  const float epsilon{1e-6};
-  return ((w > -epsilon) && (w < epsilon));
+bool __device__ near_zero<int32_t>(int32_t w, int32_t epsilon) {
+  return (w == 0);
```

Comment:
Why aren't we using epsilon here?

Reply:
For integer weights we can do an exact comparison to zero.  For floating point values we need to do an approximate comparison.